crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -317,10 +337,124 @@ pub fn extract_meshes(
     commands.insert_or_spawn_batch(entities);
 }
 
+bitflags::bitflags! {
+    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+    #[repr(transparent)]
+    pub struct MeshPipelineViewLayoutKey: u32 {
+        const MULTISAMPLED                = (1 << 0);
+        const DEPTH_PREPASS               = (1 << 1);
+        const NORMAL_PREPASS              = (1 << 2);
+        const MOTION_VECTOR_PREPASS       = (1 << 3);
+        const DEFERRED_PREPASS            = (1 << 4);
+    }
+}
+
+impl MeshPipelineViewLayoutKey {
+    const COUNT: usize = Self::all().bits() as usize + 1;
+
+    /// Builds a unique label for each layout based on the flags
+    pub fn label(&self) -> String {
+        format!(
+            "mesh_view_layout{}{}{}{}{}",
+            if self.contains(MeshPipelineViewLayoutKey::MULTISAMPLED) {
+                "_multisampled"
+            } else {
+                ""
+            },
+            if self.contains(MeshPipelineViewLayoutKey::DEPTH_PREPASS) {
+                "_depth"
+            } else {
+                ""
+            },
+            if self.contains(MeshPipelineViewLayoutKey::NORMAL_PREPASS) {
+                "_normal"
+            } else {
+                ""
+            },
+            if self.contains(MeshPipelineViewLayoutKey::MOTION_VECTOR_PREPASS) {
+                "_motion"
+            } else {
+                ""
+            },
+            if self.contains(MeshPipelineViewLayoutKey::DEFERRED_PREPASS) {
+                "_deferred"
+            } else {
+                ""
+            },
+        )
+    }
```

Comment:
Honestly I feel like long-term a macro would be good here because having to generate these strings at runtime introduces overhead. But a lot of stuff is waiting on this PR so I won't block it.

Reply:
These should only be generated once at startup, so realistically the impact will be minimal. For 32 strings of this length, it should also be < 1kb of extra memory consumption, so not a big deal IMO either.